### ISSUE TITLE
Fix `vtk` segmentation errors in integration tests or only support Trame for VTK 9.2+

### ISSUE THREAD

**Issue by tkoyama010** [pyvista/pyvista#5334]
### Describe what maintenance you would like added.

Our CI frequently gets `vtk` segmentation errors during unit test runs. We have to re-run with "re-run failed job" to make the test pass, but this is manual and very tedious. We need to fix `vtk` segmentation errors in integration tests or only support Trame for VTK 9.2+.

This is an ongoing issue. Let's report in this thread about CI environment etc when we have reruns error.

### Links to source code.

https://github.com/pyvista/pyvista/blob/main/.github/workflows/integration-tests.yml

### Pseudocode or Screenshots

None

**Comment by beroda** [comment]
To help diagnose flaky tests, the [`pytest-repeat`](https://github.com/pytest-dev/pytest-repeat) plugin might be useful.
```python
import pytest
pytestmark = pytest.mark.repeat(3)

def test_one():
    ...

@pytest.mark.repeat(6)
def test_two():
    ...
```
(My two cents)

**Comment by tkoyama010** [comment]
> To help diagnose flaky tests, the [`pytest-repeat`](https://github.com/pytest-dev/pytest-repeat) plugin might be useful.
> 
> ```python
> import pytest
> pytestmark = pytest.mark.repeat(3)
> 
> def test_one():
>     ...
> 
> @pytest.mark.repeat(6)
> def test_two():
>     ...
> ```
> 
> Moreover, [`pytest` doc](https://docs.pytest.org/en/7.1.x/explanation/flaky.html) has good resources concerning flaky tests
> 
> (My two cents)

Thanks for the comment. I would like to +1 this idea. This is because it fulfills my following request.
1. It can be used for local unstable testing.
1. No need to make changes to GitHub Actions.
1. No need to make changes to the test target by the reason of flaky test. I may only adopt Trame for VTK 9.2+ in the future, but I want to separate that discussion.

**Comment by tkoyama010** [comment]
`tests/plotting/test_widgets.py::test_affine_widget` in Linux Unit Testing (3.10, 9.2.2) raise vtk error happen in #5336. We should list it as flaky.

**Comment by MatthewFlamm** [comment]
> `tests/plotting/test_widgets.py::test_affine_widget` in Linux Unit Testing (3.10, 9.2.2) raise vtk error happen in #5336. We should list it as flaky.

I'm wondering if this is a vtk 9.2.2 problem as it always seems to happen in that particular environment.  I wonder whether it would be better to use the latest patch release from that version instead.  But, I'm also fine keeping CI as is since it is only 1 flaky test and addressing the test itself somehow.

**Comment by tkoyama010** [comment]
> > `tests/plotting/test_widgets.py::test_affine_widget` in Linux Unit Testing (3.10, 9.2.2) raise vtk error happen in #5336. We should list it as flaky.
> 
> I'm wondering if this is a vtk 9.2.2 problem as it always seems to happen in that particular environment. I wonder whether it would be better to use the latest patch release for vtk for 9.2. But, I'm also fine keeping CI as is since it is only 1 flaky test and addressing the test itself somehow.

Yes, I do not intend to change the CI environment itself. In my mind, I want to run pytest as suggested above, with multiple runs of flaky tests. To do this, we are listing the cases where errors are occurring. If you like, please share in this thread when the error occurs.

**Comment by tkoyama010** [comment]
New error happens in [Linux Unit Testing (3.10, 9.2.2)](https://github.com/pyvista/pyvista/actions/runs/7204624222/job/19626361240#logs). Also the error from `tests/plotting/test_widgets.py::test_affine_widget`.

**Comment by banesullivan** [comment]
[Linux Unit Testing (3.9, 9.1)](https://github.com/pyvista/pyvista/actions/runs/7242386255/job/19727742976?pr=5360#logs): `tests/plotting/jupyter/test_trame.py::test_trame_plotter_ui[vue2] Fatal Python error: Segmentation fault`

**Comment by tkoyama010** [comment]
[Linux Unit Testing (3.9, 9.1)](https://github.com/pyvista/pyvista/actions/runs/7258833538/job/19774819545?pr=5332#logs): `tests/plotting/jupyter/test_trame.py::test_trame_plotter_ui[vue2] Fatal Python error: Segmentation fault`

**Comment by user27182** [comment]
Getting some intermittent failures with some `download` docstrings lately, e.g. [docstring check](https://github.com/pyvista/pyvista/actions/runs/8208564075/job/22452240061?pr=5747#step:8:2243) failure:
```
UNEXPECTED EXCEPTION: ReadTimeout(ReadTimeoutError("HTTPSConnectionPool(host='raw.githubusercontent.com', port=443): Read timed out. (read timeout=5)"))
```

maybe there's a way to increase the timeout?

**Comment by banesullivan** [comment]
Going to close this due to staleness and say lets create new issues to track flaky tests as they come up
